fbpic/particles/elementary_process/ionization/ionizer.py

```diff
@@ -288,13 +285,14 @@ def handle_ionization( self, ion ):
         assert len(self.target_species) == n_levels
         for i_level, elec in enumerate(self.target_species):
             old_Ntot = elec.Ntot
-            new_Ntot = old_Ntot + cumulative_n_ionized[i_level,-1]
+            # Cast to int transfers the data from the GPU if needed
+            new_Ntot = old_Ntot + int( cumulative_n_ionized[i_level,-1] )
```

Comment:
The second memcpy is here - the new number of particles `new_Ntot` depends on the result of the `cumsum`, now on the GPU. It has to be copied to the CPU in order to allocate the new arrays, which is done by the cast to `int`.